### PR TITLE
[release/6.0.3xx] Backport Improve console output of `dotnet watch` ⌚ (#23318)

### PR DESCRIPTION
**Pull request by TanayParikh**
(cherry picked from commit 5ae51314f1d8602854c65550d4541ada686672e6)

Backports https://github.com/dotnet/sdk/pull/23318

### CODE REVIEW

**Review by pranavkm — APPROVED** [review]
Approving backport change

**Comment by TanayParikh** [comment]
@DamianEdwards just a fyi we may run into https://github.com/Microsoft/Terminal/issues/190.

**Comment by DamianEdwards** [comment]
@TanayParikh have you tried it in latest Windows Terminal? How does it look?

**Comment by TanayParikh** [comment]
> @TanayParikh have you tried it in latest Windows Terminal? How does it look?

Pranav tried out emoji support it works fine in the latest Windows Terminal (powershell, cmd). The issue is with the old conhost based command prompt. I'll do a quick validation round in the new terminal with the dotnet watch utility before merging.

I'm also updating the PR to add a `DOTNET_WATCH_SUPPRESS_EMOJIS` environment variable which to facilitate configurability.

The bunch of commits here are to do with some strange CI behavior between main and release/6.0.2xx (for some reason emoji encoding isn't working as expected in the 6.0.2xx CI).

**Comment by TanayParikh** [comment]
> I'll do a quick validation round in the new terminal with the dotnet watch utility before merging.

image

image


Looking good.

File: src/BuiltInTools/dotnet-watch/DotNetWatchOptions.cs [hunk 1]
```diff
@@ -18,6 +19,7 @@ public record DotNetWatchOptions(
             SuppressMSBuildIncrementalism: IsEnvironmentSet("DOTNET_WATCH_SUPPRESS_MSBUILD_INCREMENTALISM"),
             SuppressLaunchBrowser: IsEnvironmentSet("DOTNET_WATCH_SUPPRESS_LAUNCH_BROWSER"),
             SuppressBrowserRefresh: IsEnvironmentSet("DOTNET_WATCH_SUPPRESS_BROWSER_REFRESH"),
+            SuppressEmojis: IsEnvironmentSet("DOTNET_WATCH_SUPPRESS_EMOJIS"),
```
Comment (pranavkm) [inline src/BuiltInTools/dotnet-watch/DotNetWatchOptions.cs:22]: @DamianEdwards suggestions?

**Review by pranavkm — APPROVED** [review]

File: src/BuiltInTools/dotnet-watch/DotNetWatchOptions.cs (hunk 1 above)
Comment (DamianEdwards) [reply src/BuiltInTools/dotnet-watch/DotNetWatchOptions.cs:22]: Name is good

**Review by DamianEdwards — APPROVED** [review]

**Pull request merged**